contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -1251,6 +1264,60 @@ SendColumnMetadataToken(int natts, bool sendRowStat)
 			}
 		}
 
+		//For Spatial Types
+			tup = SearchSysCache1(TYPEOID, ObjectIdGetDatum(col->pgTypeOid));  //Search SysCache for type name
+			typTup = (Form_pg_type) GETSTRUCT(tup);
+
+			/*
+			 * Check if it is spatial Data type
+			 * Send the Corresponding MetaData Columns
+			*/			
+			if(strcmp(NameStr(typTup->typname),"geometry") == 0 || strcmp(NameStr(typTup->typname),"geography") == 0)
+			{
+				//Driver Expects assembly Name and Length 
+				//TODO: For Now assembly Name is Hardcoded but can be Altered in future
+				if(strcmp(NameStr(typTup->typname),"geometry") == 0){
+					assemblyName = "Microsoft.SqlServer.Types.SqlGeometry, Microsoft.SqlServer.Types, Version=11.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91";
+				}
+				else
+				{
+					assemblyName = "Microsoft.SqlServer.Types.SqlGeography, Microsoft.SqlServer.Types, Version=11.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91";
+				}
+
+				//Current Database Name and Length are expected by the Driver
+				db_name = pltsql_plugin_handler_ptr->get_cur_db_name();
+				temp8 = (uint8_t) pg_mbstrlen(db_name);
+				resetStringInfo(&tempBuf);
+				TdsUTF8toUTF16StringInfo(&tempBuf, db_name,
+										strlen(db_name));
+				TdsPutbytes(&temp8, sizeof(temp8));
+				TdsPutbytes(tempBuf.data, tempBuf.len);
+
+				//Since Schema Name is always sys in Babelfish Server we can directly send it
```

Comment:
what about user-defined schema?

Reply:
We are Providing only Data Type Schema and not the User Defined Schema, So that's why it's hardcoded to "sys"